h2o-core/src/main/java/water/util/ArrayUtils.java

```diff
@@ -667,6 +667,26 @@ public static String arrayToString(int[] ary) {
     return result;
   }
 
+  public static String toStringQuotedElements(Object[] a) {
+    if (a == null)
+      return "null";
+
+    int iMax = a.length - 1;
+    if (iMax == -1)
+      return "[]";
+
+    StringBuilder b = new StringBuilder();
+    b.append('[');
+    for (int i = 0; ; i++) {
+      b.append('"');
+      b.append(a[i]);
+      b.append('"');
+      if (i == iMax)
+        return b.append(']').toString();
```

Comment:
It might be my personal preference, but this return inside-if-inside-for is not very readable to me.
also the missing end condition in the for loop
I would suggest replacing this if with `if (i != iMax) b.append(", ");` and moving the return to the end of method

Reply:
Since this is not a performance critical part of code, my two cents to:

```public static String toStringQuotedElements(Object[] a) {
    if (a == null)
      return "null";
    if (0 == a.length)
      return "[]";
    return "[\"" + String.join("\",\"", new String [] { "foo", "bar" }) + "\"]";
  }```